src/rules/noUnnecessaryBindRule.ts

```diff
@@ -0,0 +1,151 @@
+            boundExpression.getText(context.sourceFile));
+
+        context.addFailureAtNode(node, Rule.FAILURE_STRING_ARROW, replacement);
+    }
+
+    function checkFunctionExpression(callExpression: ts.CallExpression, valueDeclaration: ts.FunctionExpression): void {
+        if (!canFunctionExpressionBeFixed(callExpression, valueDeclaration)) {
+            return;
+        }
+
+        const replacement = Lint.Replacement.replaceNode(
+            callExpression,
+            Lint.convertFunctionToArrowText(valueDeclaration, context.sourceFile));
+
+        context.addFailureAtNode(callExpression, Rule.FAILURE_STRING_FUNCTION, replacement);
+    }
+
+    function getArrowFunctionDeclaration(node: ts.Node): ts.ArrowFunction | undefined {
+        if (typeChecker === undefined) {
+            return undefined;
+        }
+
+        const { symbol } = typeChecker.getTypeAtLocation(node);
+        if (symbol === undefined) {
+            return undefined;
+        }
+
+        const { valueDeclaration } = symbol;
+        if (valueDeclaration === undefined) {
+            return undefined;
+        }
+
+        if (!tsutils.isArrowFunction(valueDeclaration)) {
+            return undefined;
+        }
+
+        return valueDeclaration;
+    }
+
+    function checkCallExpression(node: ts.CallExpression): void {
+        const bindExpression = node.expression;
+        if (!isBindPropertyAccess(bindExpression)) {
+            return;
+        }
+
+        const boundExpression = Lint.unwrapParentheses(bindExpression.expression);
+        if (tsutils.isFunctionExpression(boundExpression)) {
+            checkFunctionExpression(node, boundExpression);
+            return;
+        }
+
+        const valueDeclaration = getArrowFunctionDeclaration(boundExpression);
+        if (valueDeclaration !== undefined) {
+            checkArrowFunction(node, boundExpression);
+        }
+    }
+
+    return ts.forEachChild(context.sourceFile, function callback(node: ts.Node): void {
+        if (ts.isCallExpression(node)) {
+            checkCallExpression(node);
+        }
+
+        return ts.forEachChild(node, callback);
+    });
+}
+
+function isBindPropertyAccess(node: ts.LeftHandSideExpression): node is ts.PropertyAccessExpression {
+    return ts.isPropertyAccessExpression(node) && node.name.text === "bind";
+}
+
+function canFunctionExpressionBeFixed(callExpression: ts.CallExpression, valueDeclaration: ts.FunctionExpression): boolean {
+    if (callExpression.arguments.length !== 1
+        || callExpression.arguments[0].kind !== ts.SyntaxKind.ThisKeyword
+        || valueDeclaration.asteriskToken !== undefined) {
+        return false;
+    }
+
+    const { name } = valueDeclaration;
+    if (name === undefined) {
+        return true;
+    }
+
+    return !ts.forEachChild(
+        valueDeclaration.body,
+        function search(node: ts.Node): boolean {
+            if (tsutils.isIdentifier(node)) {
+                return node.text === name.text;
```

Comment:
This could yield a lot of false negatives. Consider using `collectVariableUsage` from tsutils